### ISSUE TITLE
`about` errors on structs with fields that have no size

### ISSUE THREAD

**Issue by darsnack** [tecosaur/About.jl#1]
Currently, calling `about(somestruct)` is broken for fields that are functions:
```julia
julia> struct Foo
           a::Int
           b::Matrix{Float32}
           c::typeof(identity)
       end

julia> f = Foo(1, rand(3, 4), identity)
Foo(1, Float32[0.8633268 0.40085334 0.27201754 0.60436267; 0.34008977 0.2632495 0.82902575 0.6264377; 0.32734197 0.008732563 0.8294697 0.10690562], identity)

julia> about(f)
Foo, 16B referencing 128B (<: Any)
 a::Int64            8B 000000000000000000000000 … 00000000000000000000001 1
 b::Matrix{Float32}  8B Ptr{Float32} @0x000000010cbac5c0                   Float32[0.8 … 7 0.106906]
 c::typeof(identity) 0B Ptr?                                               identity

ERROR: InexactError: Int64(NaN)
Stacktrace:
 [1] Int64
   @ ./float.jl:981 [inlined]
 [2] convert
   @ ./number.jl:7 [inlined]
 [3] round
   @ ./rounding.jl:469 [inlined]
 [4] round
   @ ./rounding.jl:467 [inlined]
 [5] memorylayout(io::Base.TTY, type::DataType)
   @ About ~/.julia/packages/About/9q1TE/src/types.jl:102
 [6] memorylayout(io::Base.TTY, value::Foo)
   @ About ~/.julia/packages/About/9q1TE/src/values.jl:72
 [7] about(io::Base.TTY, value::Foo)
   @ About ~/.julia/packages/About/9q1TE/src/values.jl:18
 [8] about(x::Foo)
   @ About ~/.julia/packages/About/9q1TE/src/About.jl:24
 [9] top-level scope
   @ REPL[12]:1
```

It seems that the issue is that functions have a "size" of zero.
```julia
julia> About.structinfo(typeof(f))
3-element Vector{About.FieldInfo}:
 About.FieldInfo(1, :bright_green, 0, 8, 8, false, :a, Int64)
 About.FieldInfo(2, :bright_yellow, 8, 8, 0, true, :b, Matrix{Float32})
 About.FieldInfo(3, :bright_magenta, 16, 0, 0, true, :c, typeof(identity))
```

**Comment by giordano** [comment]
Probably related:
```
julia> about(nothing)
Nothing, 0B (<: Any)
ERROR: ArgumentError: reducing over an empty collection is not allowed; consider supplying `init` to the reducer
```

**Comment by tecosaur** [comment]
This also overlaps with #5, I'll try to do a round of improvements next week :)

**Comment by tecosaur** [comment]
The handling of `about(nothing)` should be fixed in 429bd1161f0ea5038495527b5be8af9df5ea18fe

The handling of `about(f::Foo)` raised here should be fixed in bb8a4a8bf67135e2934abdc9709d9cea60c8a34e.

**Comment by tecosaur** [comment]
For reference:

![image](https://github.com/tecosaur/About.jl/assets/20903656/2cf7143a-ac33-4710-9dae-32ec0552c733)
